### PR TITLE
🏃Test conversions with object fuzzer

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: api/v1alpha2/awsmachine_conversion.go [hunk 1]
```diff
@@ -51,10 +52,13 @@ func restoreAWSMachineSpec(restored *infrav1alpha3.AWSMachineSpec, dst *infrav1a
 	}
 }
 
+func restoreAWSMachineStatus(restored *infrav1alpha3.AWSMachineStatus, dst *infrav1alpha3.AWSMachineStatus) {
+	dst.FailureReason = restored.FailureReason
```
Comment (detiber) [inline api/v1alpha2/awsmachine_conversion.go]: Shouldn't FailureMessage/FailureReason be migrated to ErrorMessage/ErrorReason in v1alpha2?

File: api/v1alpha2/awsmachine_conversion.go [hunk 2]
```diff
@@ -36,13 +36,14 @@ func (src *AWSMachine) ConvertTo(dstRaw conversion.Hub) error { // nolint
...
 	}
-
 	restoreAWSMachineSpec(&restored.Spec, &dst.Spec)
+	restoreAWSMachineStatus(&restored.Status, &dst.Status)
 
 	return nil
 }
 
 func restoreAWSMachineSpec(restored *infrav1alpha3.AWSMachineSpec, dst *infrav1alpha3.AWSMachineSpec) {
+	dst.FailureDomain = restored.FailureDomain
```
Comment (detiber) [inline api/v1alpha2/awsmachine_conversion.go]: Shouldn't this be migrated to/from AvailabilityZone in v1alpha2?

Comment (vincepri) [reply api/v1alpha2/awsmachine_conversion.go]: Yes! Found the issue we had with the current conversion!

File: api/v1alpha2/awsmachine_conversion.go (hunk 1 above)
Comment (vincepri) [reply api/v1alpha2/awsmachine_conversion.go]: my bad :)

File: api/v1alpha2/awscluster_conversion.go [hunk 3]
```diff
@@ -44,8 +46,15 @@ func (src *AWSCluster) ConvertTo(dstRaw conversion.Hub) error { // nolint
 	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
 		return err
 	}
+
 	dst.Spec.ImageLookupOrg = restored.Spec.ImageLookupOrg
-	dst.Spec.ControlPlaneLoadBalancer.CrossZoneLoadBalancing = restored.Spec.ControlPlaneLoadBalancer.CrossZoneLoadBalancing
+	dst.Spec.ImageLookupBaseOS = restored.Spec.ImageLookupBaseOS
+	if restored.Spec.ControlPlaneLoadBalancer != nil {
+		restored.Spec.ControlPlaneLoadBalancer.DeepCopyInto(dst.Spec.ControlPlaneLoadBalancer)
```
Comment (ncdc) [inline api/v1alpha2/awscluster_conversion.go]: Why not `=` assignment - the restored data came from unmarhshaling the json in the annotation, not from the cache?

Comment (vincepri) [reply api/v1alpha2/awscluster_conversion.go]: Thanks, I changed it last night and forgot to change it back

**Comment by vincepri** [comment]
This should be ready @ncdc

**Comment by randomvariable** [comment]
lgtm

**Pull request merged**